### PR TITLE
Understanding Diffie Helman and Elgamal Asymmetric Encryption

### PR DESCRIPTION
**Pull request by njerigitahi**

### CODE REVIEW

File: content/articles/diffie-hellman-elgamal/index.md [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+2. Creation and verification of the digital signature.
+   - If the use of cryptography were to become widespread, not just in military situations but for commercial and private purposes, then electronic messages and documents would need the equivalent of signatures used in paper documents.
+
+>Asymmetric algorithms rely on one key for encryption and a different but related key for decryption.
+
+These algorithms have the following important characteristic:
+1. It is computationally infeasible to determine the decryption key, given only knowledge of the cryptographic algorithm and the encryption key.
+2. The two related keys can be used for encryption, with the other used for decryption.
+
+![Asymmetric encryption.](/\engineering-education\content\articles\understanding-diffie-helman-and-elgamal-asymmetric-encryption/asymmetric-encryption.png)
```
Comment (jumamiller) [inline content/articles/diffie-hellman-elgamal/index.md:77]: Describe what this image does!

File: content/articles/diffie-hellman-elgamal/index.md [hunk 2]
```diff
@@ -0,0 +1,301 @@
...
+   p=16 mod 5=1
+A3
+ >>x remain 3
+   p=1 mod 5=1
+A4
+>> x=3 mod 5=3
+   p=1 mod 5=1
+Return the last value of x = 3
+3^9 mod 5 =3
+```
```
Comment (jumamiller) [inline content/articles/diffie-hellman-elgamal/index.md:132]: Describe what's happening on this section!

File: content/articles/diffie-hellman-elgamal/index.md [hunk 3]
```diff
@@ -0,0 +1,301 @@
...
+y = int(input("Enter Second Value :"))
+z = (int)(1e9+7) # to compute some very, very large number 
+# power function use
+d = power(x, y) % z 
+print ("Value Is =",d) 
+```
+*Output*
+Enter First Value : 2
+Enter Second Value : 3
+Value  = 8
```
Comment (jumamiller) [inline content/articles/diffie-hellman-elgamal/index.md:147]: Explain this code snippet!

File: content/articles/diffie-hellman-elgamal/index.md [hunk 4]
```diff
@@ -0,0 +1,301 @@
...
+
+### Working of DHKE Protocol
+It involves exchanging secret/asymmetric keys between sender and receiver by using asymmetric encryption(public and private key)
```
Comment (jumamiller) [inline content/articles/diffie-hellman-elgamal/index.md:150]: Could explain the DHKE protocol?

File: content/articles/diffie-hellman-elgamal/index.md [hunk 5]
```diff
@@ -0,0 +1,301 @@
...
+
+#### Steps involved
+1. Assume a prime number `Q`
+2. Set `B` a primitive root of `Q`
+ `B < Q`
```
Comment (jumamiller) [inline content/articles/diffie-hellman-elgamal/index.md:155]: Steps involved to accomplish what?

**Review by jumamiller — CHANGES_REQUESTED** [review]
Hello @njerigitahi, kindly work on the suggestions!

File: content/articles/diffie-hellman-elgamal/index.md [hunk 6]
```diff
@@ -0,0 +1,301 @@
...
+date: 2021-09-05T00:00:00-02:00 
+topics: [Security]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/understanding-diffie-helman-and-elgamal-asymmetric-encryption/hero.jpg
+    alt: Encryption Example Image
+---
+Security plays a crucial role in data communication and data protection. It helps prevent unauthorized access to confidential information, which may lead to data loss and data modification by unknown people making data transferred safe and unchanged.
+<!--more-->
```
Comment (jumamiller) [inline content/articles/diffie-hellman-elgamal/index.md:17]: The second sentence in this paragraph is too long and needs to be broken down.

**Review by njerigitahi — COMMENTED** [review]
@jumamiller, changes done!

**Review by jumamiller — APPROVED** [review]
@njerigitahi, you have made progress updating this article!

**Comment by Kayere** [comment]
Congratulations on publishing your first article @njerigitahi. Please go through the changes made by your peer reviewer and I to help improve the quality of your future articles.

Thank you for the review @jumamiller.

**Pull request merged**